### PR TITLE
Add Command::executeCommand()

### PR DESCRIPTION
**Pull request by markstory**
This method enables commands to dispatch each other without needing to use the CommandRunner. This provides commands with the ability to be composed from other commands.

Refs #12549

### CODE REVIEW

**Review by themrwilliams — APPROVED** [review]

File: src/Console/Command.php [hunk 1]
```diff
@@ -249,4 +249,34 @@ public function abort($code = self::CODE_ERROR)
...
+
+    /**
+     * Execute another command with the provided set of arguments.
+     *
+     * @param string|\Cake\Console\Command $command The command class name or command instance.
+     * @param \Cake\Console\ConsoleIo $io The ConsoleIo instance to use for the executed command.
+     * @param array $args The arguments to invoke the command with.
+     * @return int|null The exit code or null for success of the command.
+     */
+    public function executeCommand($command, ConsoleIo $io, array $args = [])
```
Comment (ADmad) [inline src/Console/Command.php]: Can the method name and arguments order be harmonized with `CommandRunner::runCommand()`?
Could the `$io` argument be made optional and internally create a new instance if none provided?

Comment (markstory) [reply src/Console/Command.php]: I could do that. Making a new instance will do bad things in tests which is why I made it a required argument.

Comment (markstory) [reply src/Console/Command.php]: Done now :+1:

File: src/Console/Command.php [hunk 2]
```diff
@@ -249,4 +249,35 @@ public function abort($code = self::CODE_ERROR)
...
+
+    /**
+     * Execute another command with the provided set of arguments.
+     *
+     * @param string|\Cake\Console\Command $command The command class name or command instance.
+     * @param array $args The arguments to invoke the command with.
+     * @param \Cake\Console\ConsoleIo $io The ConsoleIo instance to use for the executed command.
+     * @return int|null The exit code or null for success of the command.
+     */
+    public function executeCommand($command, array $args = [], ?ConsoleIo $io = null)
```
Comment (ADmad) [inline src/Console/Command.php]: `?ConsoleIo` typehint won't work in PHP < 7.1 :stuck_out_tongue:.

Comment (dereuromark) [reply src/Console/Command.php]: Dropping the ? should be fine here :)

Comment (markstory) [reply src/Console/Command.php]: Derp. I forgot that this was for 3.x

**Comment by markstory** [comment]
> Fatal error: Declaration of Cake\TestSuite\TestCase::setUp() must be compatible with PHPUnit\Framework\TestCase::setUp(): void in /home/travis/build/cakephp/cakephp/src/TestSuite/TestCase.php on line 36

Going to rollback the PHPUnit version until we can make a separate effort to do that.

**Review by ADmad — APPROVED** [review]

**Pull request merged**